### PR TITLE
Investigate why zpools become un-importable at the end of tests

### PR DESCRIPTION
**Pull request by tanabarr**
During debug of #244 it became obvious that zpools get to a state where they cannot be  exported and imported between SSI tests.

We need to investigate why this occurs and where potential unmounting of targets fails.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-hpdd/intel-manager-for-lustre/289)

### CODE REVIEW

**Comment by brianjmurrell** [comment]
> potentially related to https://jira.hpdd.intel.com/browse/HYD-3468

These seem like very different, unrelated issues to me.  Can you expand on how you think they might be related?

**Comment by tanabarr** [comment]
fetch_help enabled capturing state where pool is not importable:
test log shows failed export from all hosts.

Zpool on disk13 shows as UNAVAIL and imported on node "8":
```
[root@lotus-55vm16 ~]#  zpool import
   pool: zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk13
     id: 10105200814420078163
  state: UNAVAIL
 status: The pool is currently imported by another system.
 action: The pool must be exported from lotus-55vm18 (hostid=5983c94a)
        before it can be safely imported.
   see: http://zfsonlinux.org/msg/ZFS-8000-EY
 config:

        zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk13  UNAVAIL  currently in use
          scsi-0QEMU_QEMU_HARDDISK_disk13  ONLINE

   pool: zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk11
     id: 4906354292184478512
  state: UNAVAIL
 status: The pool is currently imported by another system.
 action: The pool must be exported from lotus-55vm15 (hostid=60ab4aa6)
        before it can be safely imported.
   see: http://zfsonlinux.org/msg/ZFS-8000-EY
 config:

        zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk11  UNAVAIL  currently in use
          scsi-0QEMU_QEMU_HARDDISK_disk11  ONLINE

   pool: zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk15
     id: 7226628311757996711
  state: ONLINE
 action: The pool can be imported using its name or numeric identifier.
 config:

        zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk15  ONLINE
          scsi-0QEMU_QEMU_HARDDISK_disk15  ONLINE
```

Host "8" fails to export because the target on zpool is still mounted:
```
zfs_pool_scsi0QEMU_QEMU_HARDDISK_disk13/testfs-OST0000 on /mnt/testfs-OST0000 type lustre (ro)
```

**Comment by tanabarr** [comment]
so a second umount call in zfs_cleanup between tests seems to make zpools successfully export as desired

File: chroma-manager/tests/integration/core/api_testcase_with_test_reset.py [hunk 1]
```diff
@@ -912,6 +912,9 @@ def cleanup_zfs_pools(self, test_servers, action, zpool_datasets, devices_must_e
         # Ensure agents stopped to avoid interference with pool imports/exports
         self.remote_operations.stop_agents([server['fqdn'] for server in test_servers])
 
+        # Attempt to unmount all lustre targets otherwise we won't be able to export parent pool
+        [self.remote_operations.unmount_lustre_targets(server) for server in test_servers]
```
Comment (brianjmurrell) [inline chroma-manager/tests/integration/core/api_testcase_with_test_reset.py]: This kind of unorderly unmounting of lustre targets (i.e. as opposed to a proper filesystem stop such as `ApiTestCaseWithTestReset.stop_filesystem()`) is what has led to #246.

Additionally, unmounting targets from underneath Pacemaker is particularly bad.  Is there any guarantee that Pacemaker is not running before we do this `unmount_lustre_targets`?

Comment (tanabarr) [reply chroma-manager/tests/integration/core/api_testcase_with_test_reset.py]: yes, it's after clear_ha

Comment (brianjmurrell) [reply chroma-manager/tests/integration/core/api_testcase_with_test_reset.py]: But `clear_ha()` does it's own [`unmount_lustre_targets()`](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/pool-unimportable/chroma-manager/tests/integration/core/remote_operations.py#L1232).  Maybe this is related to #246 and now that that is merged into this branch, you can remove this?

**Comment by tanabarr** [comment]
#325 contain the majority of commits aimed at avoiding the pool corruptions

**Pull request closed without merging**